PORT-15403 Added migration guide to gitlab (#2526)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/git/gitlab/migration.md

---
sidebar_position: 6
---

# Migration guide to the v2 integration

> This guide helps you migrate from Port's GitLab integration to the new GitLab-v2 integration.

Reviewer: ```suggestion
This guide walks through the process of migrating from Port's GitLab integration to the new GitLab-v2 integration.
```

| **folder** kind | Simple repo names + single branch | Repository objects with name/branch pairs | Update repos to objects with `name` and `branch` properties |
| **pipeline** kind | ❌ Not available | ✅ New kind | Add pipeline mapping if needed |
| **job** kind | ❌ Not available | ✅ New kind | Add job mapping if needed |
| **issue** kind | ❌ Limited support | ✅ Enhanced | Update to new issue mapping |

### File and folder mapping updates

The way you specify repositories for file and folder resync has changed significantly between versions.

| Repository Specification | GitLab Format | GitLab-v2 Format |
|--------------------------|-----------|-----------|
| File repos | `"backend-service"` | `"my-org/backend-service"` |
| Folder repos | `"frontend-app"` | `"my-org/frontend-app"` |

The namespace path format follows GitLab's standard `group/project` or `group/subgroup/project` structure, ensuring accurate repository identification even when multiple projects share the same name across different groups.

> ### Migration steps

Reviewer: the headers are still a bit off, only the changes should be under the "Changes" header. The "Migration steps" for example, should be in the same level (## and not ###) since its a separate point.

:::

#### Step 7: Clean up GitLab integration

Once GitLab-v2 is working correctly:

1. **Disable the GitLab integration**
2. **Revoke the old group access tokens** 
3. **Remove GitLab configuration files**
4. **Update any documentation** referencing the old setup

This cleanup ensures security by removing unused tokens and prevents confusion from having multiple configurations.

### New features in GitLab-v2

#### Pipeline jobs support

GitLab-v2 introduces comprehensive pipeline job tracking, allowing you to monitor CI/CD job status, stages, and duration. This feature provides detailed insight into your build and deployment processes.

```yaml
- kind: job
  selector:
    query: 'true'
  port:
    entity:
      mappings:
        identifier: .id | tostring
        title: .name
        blueprint: '"gitlabJob"'
        properties:
          status: .status
          stage: .stage
          duration: .duration
```

#### Enhanced group management

Groups in GitLab-v2 provide cleaner member resync and better hierarchy handling. The separation of member and group concerns allows for more flexible data modeling and improved relationship management between users, groups, and projects.

> ## Troubleshooting common issues

Reviewer: ```suggestion
## Troubleshoot common issues
```
